### ISSUE TITLE
2019-12-11 M4SS computation error when all concentrations are zero

### ISSUE THREAD

**Issue by jhhughes256** [tensfeldt/openNCA#102]
### Error

When running tc008 M4SS, tc018 M4SS, tc105 M4SS with all concentrations are set to zero the following computation error occurs. The below example was for tc105 M4SS, however error messages are similar between test cases.

```
Error in value[[3L]](cond) : 
  Error in `[<-.data.frame`(`*tmp*`, i, , value = c(661083556, 0, 0, NA, : replacement has 35 items, need 39
For SDEID 661083556 
7. stop(paste0(e, "For SDEID ", unique(data_data[, map_data$SDEID])[i])) at run_M4_SS_computation.R#1518
6. value[[3L]](cond) 
5. tryCatchOne(expr, names, parentenv, handlers[[1L]]) 
4. tryCatchList(expr, classes, parentenv, handlers) 
3. tryCatch({
    if (comp_required[["DIi"]]) {
        di <- list()
    } ... at run_M4_SS_computation.R#688
2. run_M4_SS_computation(data = merged_data, map = map_data, method = method, 
    model_regex = model_regex, parameter_list = parameter_list, 
    return_list = return_list) at run_computation.R#2390
1. run_computation(data = d, map = mct, flag = flags, parameterset = parameterset) 
```

The parameterset for each of these test cases is included below in case they are needed for replicating this issue.

#### tc008 M4SS

```
"AE;AEPCT;AET;AETPCT;AT;AET;AETAU;AETAUi;AURCALL;AURCINFO;AURCINFP;AURCLAST;AURCT;AURCT1_T2;AURCXPCTO;AURCXPCTP;DOSE1;KEL;KELNOPT;KELR;KELRSQ;KELRSQA;KELTMHI;KELTMLO;MAXRATE;MIDPTLAST;MIDPTA;MIDPTN;RATEA;RATELAST;RATEN;THALF;THALFF;TLAG;TMAXRATE;VOLSUM;FLGACCEPTKEL;FLGACCEPTPREDOSE;FLGACCEPTTAU;FLGACCEPTTMAX;FLGACCEPTKELCRIT;FLGACCEPTPREDOSECRIT;SPANRATIOCRIT"
```

#### tc018 M4SS

```
"AE;AEPCT;AET;AETAUi;AETAUpti;AETPCT;AT;AURCALL;AURCINFO;AURCINFP;AURCLAST;AURCT;AURCT1_T2;AURCXPCTO;AURCXPCTP;CLRT;CLRTAUi;DOSEC;DIi;DOSEi;FA;KEL;KELNOPT;KELR;KELRSQ;KELRSQA;KELTMHI;KELTMLO;MAXRATE;MAXRATEi;MIDPTA;MIDPTLASTi;MIDPTN;RATEA;RATELASTi;RATEN;TAUi;THALF;THALFF;TLAG;TMAXRATE;TMAXRATEi;TOLDi;VOLSUM;FLGACCEPTKEL;FLGACCEPTPREDOSE;FLGACCEPTTAU;FLGACCEPTTMAX"
```

#### tc105 M4SS

```
"AE;AEPCT;AET;AETAUi;AETAUPTi;AETPCT;AURCALL;AURCINFO;AURCINFP;AURCLAST;AURCT;AURCT1_T2;AURCXPCTO;AURCXPCTP;CLR;CLRT;CLRTAU;DOSE;DOSEi;DOSEC;DOSECi;FA;MAXRATE;MAXRATEi;MIDPT;MIDPTA;MIDPTN;MIDPTLAST;MIDPTLASTi;RATELAST;RATELASTi;RATE;RATEi;RATEA;RATEAi;RATEN;RATENi;TAU;TAUi;TLAG;TMAXRATE;TMAXRATEi;TOLD;TOLDi;VOLSUM;FLGACCEPTPREDOSE;FLGACCEPTTMAX;FLGACCEPTKEL;FLGACCEPTTAU"
```

**Comment by jhhughes256** [comment]
### Update

As of this issue still occurs but with a different error shown below.

**Comment by jhhughes256** [comment]
As of 00cdcd3 this issue still occurs.

**Comment by rudraya** [comment]
This issue was addressed in 1893609, please let me know if it still occurs. I tested tc008, tc018, tc105 test cases and they were able to run as expected.

**Comment by jhhughes256** [comment]
As of be5ef37989 this issue no longer occurs.
